superset-frontend/src/views/CRUD/welcome/Welcome.tsx

```diff
@@ -179,7 +179,10 @@ function Welcome({ user, addDangerToast }: WelcomeProps) {
     setItem(LocalStorageKeys.homepage_collapse_state, state);
   };
 
+  const WelcomeMessageExtension = extensionsRegistry.get('welcome.message');
   const WelcomeTopExtension = extensionsRegistry.get('welcome.banner');
+  const WelcomeDataExtension = extensionsRegistry.get('welcome.data');
+  const WelcomeTableExtension = extensionsRegistry.get('welcome.table');
```

Comment:
I think the convention for these names is they should refer mostly to the place in the layout that the extension will go, without referring to the content that Preset will be putting into the extension. 

Ideally, these new components could fit under a single extension point, maybe something like "welcome.main.replacement" since it's replacing the body content of the welcome page.  The details that Preset uses this extension point to add extra data and a table isn't really relevant to the superset side.

Reply:
yeah it might make more sense to put the data/table into a single component and import. the message needs to go above the banner though... should we be renaming that one as well?

Reply:
Could the welcome message be added to the 'welcome.banner' extension point? 

(renaming extension points is _technically_ a breaking change. It's probably fine to do at this point, but we shouldn't be considering these to be easily renamable

Reply:
I'd like to honestly keep all 4 separate, as they ARE distinct items within the space and could be changed separately.

Reply:
to clarify, I'm advocating for consolidating these into two component files:
a WelcomeTop component, applied to `welcome.banner`, containing any components inserted at the top without replacing the homepage (Message and Banner)

and a HomeReplacement component, applied to `welcome.main.replacement`, containing any components that should replace the regular home page in their presence (Data and Table)

The components all have their distinct files, but the logic for showing them or not has to exist in the Preset side of things anyway, so I don't see any benefit in breaking them out into separate extension points.

Reply:
I think the value would be that if we decide that the message is annoying and we don't want it, we can remove it and leave the banner in a way that makes sense with the components, but allows us to easily plug it in again later.  Same for if we find the table view on the bottom wasn't popular but the recently viewed got a lot of attention, we could swap out the bottom component more easily.